### PR TITLE
Use ConvexPolygon to calculate polygon area in box2d_util

### LINKED ISSUE #1131: Use Polygon::area in box2d_util
### Description of the task

Right now `box2d_util` has its own implementation for finding the area of a polygon. Once the new geometry changes are finished, we should use the `Polygon::area` implementation to avoid duplication

### Acceptance criteria

- [ ] Replace the implementation of `box2d_util::polygonArea` with a call to `Polygon::area`

### Blocked By

General geom changes.

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Replaced the polygon area algorithm in box2d_util with our existing ConvexPolygon algorithm.

### Testing Done

unit tests still pass

### Resolved Issues

resolves #1131 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/simulation/physics/box2d_util.cpp [hunk 1]
```diff
@@ -47,20 +51,12 @@ float polygonArea(const b2PolygonShape& polygon)
...
-        double x_sum        = polygon.m_vertices[j].x + polygon.m_vertices[i].x;
-        double y_difference = polygon.m_vertices[j].y - polygon.m_vertices[i].y;
-        area += x_sum * y_difference;
-        j = i;
+        vertices.emplace_back(Point(polygon.m_vertices[i].x, polygon.m_vertices[i].y));
     }
+    ConvexPolygon convex_polygon(vertices);
 
-    return static_cast<float>(std::fabs(area / 2.0));
+    return static_cast<float>(convex_polygon.area());
```
Comment (jonathanlew) [inline src/software/simulation/physics/box2d_util.cpp]: nit: 
```suggestion
    return static_cast<float>(ConvexPolygon(vertices).area());
```

**Review by jonathanlew — DISMISSED** [review]
LGTM!

**Review by kelvinkoon — DISMISSED** [review]
LGTM pending existing comment.

**Review by LiCody — DISMISSED** [review]
LGTM pending other reviewers comments

File: src/software/simulation/physics/box2d_util.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/software/simulation/physics/box2d_util.cpp]: fixed

**Review by akhilveeraghanta — APPROVED** [review]

**Review by jonathanlew — APPROVED** [review]

**Review by kelvinkoon — APPROVED** [review]

**Pull request merged**